### PR TITLE
Ensure that `NoWorkResult` and `LazyResult` have the same outcome.

### PR DESCRIPTION
**Pull request by romainmenke**
see : https://github.com/postcss/postcss/issues/1869

### CODE REVIEW

File: lib/map-generator.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ class MapGenerator {
     this.root = root
     this.opts = opts
     this.css = cssString
+    this.originalCSS = cssString
```
Comment (romainmenke) [inline lib/map-generator.js:19]: There is a chicken vs. egg problem here that doesn't exist when fully parsing.

When the source is parsed there is already an `Input` instance created early. And this instance will have information about previous sourcemaps.

When the source is not parsed the `Input` instance is created in `previous()` but this is done **after** the annotation comments have already been removed.

Some mechanic is needed to get the previous sourcemaps from the original source and to store the modified css without the annotation comments.

I decided to add a second variable `this.originalCSS` which holds an unmodified copy.

I don't know if this is the best or even a good solution.
Please give feedback on this.

File: lib/map-generator.js [hunk 2]
```diff
@@ -74,7 +75,7 @@ class MapGenerator {
         }
       }
     } else if (this.css) {
-      this.css = this.css.replace(/(\n)?\/\*#[\S\s]*?\*\/$/gm, '')
+      this.css = this.css.replace(/\n*?\/\*#[\S\s]*?\*\/$/gm, '')
```
Comment (romainmenke) [inline lib/map-generator.js:78]: When fully parsed all leading whitespace before an annotation comment will be removed, not only the first `\n`.

File: lib/no-work-result.js [hunk 3]
```diff
@@ -37,6 +37,9 @@ class NoWorkResult {
       if (generatedMap) {
         this.result.map = generatedMap
       }
+    } else {
+      map.clearAnnotation();
+      this.result.css = map.css;
     }
```
Comment (romainmenke) [inline lib/no-work-result.js:43]: If no sourcemap is requested the already present inline sourcemap must also be removed.

File: lib/map-generator.js (hunk 2 above)
Comment (ahmdammarr) [reply lib/map-generator.js:78]: I don't understand how this part is related to the PR's main purpose (ensuring both `NoWorkResult` and `LazyResult` have the same outcome)

Comment (romainmenke) [reply lib/map-generator.js:78]: In a `LazyResult` the annotation is a parsed comment.
`Comment` nodes in PostCSS also contain all the leading whitespace.

In a `NoWorkResult` this RegExp is used and it only replaces a single preceding newline.

**Review by ahmdammarr — APPROVED** [review]

**Comment by ai** [comment]
Thanks. Released in 8.4.33.

**Pull request merged**